python/GafferUI/MultiLineTextWidget.py

```diff
@@ -319,9 +328,57 @@ def __drop( self, widget, event ) :
 class _PlainTextEdit( QtGui.QPlainTextEdit ) :
 
 	def __init__( self, parent = None ) :
-
 		QtGui.QPlainTextEdit.__init__( self, parent )
-
+		self.__fixedLines = None
+		self.__widgetFullyBuilt = False
+		
+	def setFixedLines( self, fixedLines ) :
+		
+		self.__fixedLines = fixedLines
+		
+		self.__computeHeight()
+
+	def getFixedLines( self ) :
+
+		return self.__fixedLines
+
+	def __computeHeight( self ) :
+		
+		# since the height can be based on the font metrics when displaying a fixed number of lines, it can be only
+		# computed properly when the stylesheet has been applied to the widget which happens when the widget 
+		# is about to be shown otherwise there is no need to compute it before that (since that font metrics
+		# would be wrong anyway)
+		if not self.__widgetFullyBuilt :
+			return
+		
+		fixedLines = self.getFixedLines()
+		
+		# when the multiline displays fixed lines
+		if fixedLines is not None :
+
+			# computing the font metrics based on the number of lines
+			height = self.fontMetrics().boundingRect( "M" ).height() * fixedLines
+			
+			# also, we need to compute the widget margins to frame the fixed lines nicely 
+			margin = self.contentsMargins().top() + self.contentsMargins().bottom() + self.document().documentMargin()
+			
+			height += margin
+			
+			self.setFixedHeight(height)
+			
+		# otherwise fallbacks to the default behavior which is auto-expanded height
+		else:
+			# resetting the height
+			QWIDGETSIZE_MAX = 16777215 # qt #define not exposed by PyQt or PySide
+			self.setFixedHeight(QWIDGETSIZE_MAX)
+		
+	def showEvent ( self, event ):
+
+		self.__widgetFullyBuilt = True		
+		self.__computeHeight()
```

Comment:
I've sent to you a test case by email, but looking at the QWidget.setSizePolicy implementation it only calls the updateGeometry when is a different size policy otherwise it returns right away (that thing is easy to get mislead), that explains why it is failing when setting a different character width for the same size policy, anyway I had to look through the Qt code again :D